Snail/Extensions/URLSessionExtension.swift

```diff
@@ -0,0 +1,63 @@
+//  Copyright © 2017 Compass. All rights reserved.
+
+import UIKit
+
+extension URLSession {
+    enum ErrorType: Error {
+        case invalidData
+        case invalidResponse
+    }
+
+    public func observable(request: URLRequest) -> Observable<[String: Any]> {
```

Comment:
this name seems weird `observable` that feels really confusing.

Why not follow the `data` pattern below?